### ISSUE TITLE
Create ABC for 

### ISSUE THREAD

**Issue by jayvdb** [bskinn/stdio-mgr#86]
stdio_mgr expects `getvalue` to exist on `BytesIO`, and previously expected it to exist on `StringIO`, and the docs still refer to StringIO.getvalue, but there is no base/abstract type which defines this.

```py
>>> import io
>>> io.BytesIO.__class__.__mro__
(<class 'type'>, <class 'object'>)
>>> io.StringIO.__class__.__mro__
(<class 'type'>, <class 'object'>)
```

There is [nothing in](https://github.com/python/typing/search?utf8=%E2%9C%93&q=getvalue) https://github.com/python/typing or [title search on bpo](https://bugs.python.org/issue?%40search_text=&ignore=file%3Acontent&title=getvalue&%40columns=title&id=&%40columns=id&stage=&creation=&creator=&activity=&%40columns=activity&%40sort=activity&actor=&nosy=&type=&components=&versions=&dependencies=&assignee=&keywords=&priority=&status=1&%40columns=status&resolution=&nosy_count=&message_count=&%40group=&%40pagesize=50&%40startwith=0&%40sortdir=on&%40action=search)

We should create our own temporary ABC for this, so we can hang features on it, add new ones (like `_PersistedFileIO` as a `BytesIO` with a `fileno()`), and raise issues upstream about it.  This is also 'needed' for bytes support (https://github.com/bskinn/stdio-mgr/issues/36) if we use strict type checking (https://github.com/bskinn/stdio-mgr/issues/60)

e.g. https://github.com/bskinn/stdio-mgr/issues/80 is a feature that depends on `.getvalue` and little else.

**Comment by jayvdb** [comment]
i.e. we need this `.getvalue` to exist on the `TextIOWrapper` instances, but it isnt defined as existing there.

**Comment by bskinn** [comment]
Agreed - we want this here, even if it's not found to be worth porting upstream into stdlib.

**Comment by jayvdb** [comment]
any naming suggestions?

**Comment by jayvdb** [comment]
AbstractIOValue?
